src/odemis/acq/path.py

```diff
@@ -259,6 +260,28 @@
 ALIGN_MODES = {'mirror-align', 'chamber-view', 'fiber-align', 'streak-align', 'spec-focus', 'spec-fiber-focus', 'streak-focus'}
 
 
+class OneTaskExecutor(ThreadPoolExecutor):
+    """
+    Only one task at a time is executed. If several tasks are queued, only the
+    last one is executed.
+    """
+
+    def __init__(self, thread_name_prefix=''):
+        super(OneTaskExecutor, self).__init__(max_workers=1, thread_name_prefix=thread_name_prefix)
+
+    def submit(self, fn, *args, **kwargs):
+        # Cancels all the previous tasks which didn't start yet
+        # Note: in practice, there should be only one task at most queued.
+        while True:
+            try:
+                self._work_queue.get(block=False)
+                logging.debug("Canceling 1 task before starting the next one")
+            except queue.Empty:
+                break
```

Comment:
You empty the queue correct? How do you then execute the last one on the queue?

Reply:
Yes, first empty the queue (so only the task which is already running is left), and then the call to the super() will add this new function to the work_queue.
I've added a comment now (and change to user `super()`).